### ISSUE TITLE
Error occurs when displaying German RichText from Storyblok & Partial handling of HexColor to Flutter Color conversion

### ISSUE THREAD

**Issue by joshthedevelopa** [alsterverse/flutter_storyblok#39]
## Bug Description
When using the internationalization, upon switching from English to German upon API call, `StoryblokRichText` returns an error and does not render widget and partial edge case handling of color conversion in `_getColorFromHex` function

## Steps to Reproduce
1. On Storyblok, create a story with one field with the type `RichText`.
2. Enable **English** and **German** as supported languages in the **Space** setting of your space.
3. Integrate a switch functionality in mobile to enable switching and querying between two languages.
4. Use the `StoryblokRichText` widget to render the `RichText` content return from the client.
5. Switch to **German** in the app

## Expected Behavior
Render content as seen for English version

## Actual Behavior
Renders an error:

## Screenshots/Logs
![Screenshot](https://github.com/user-attachments/assets/d177988b-2381-49d9-a854-d24e5b2a6efe)
Error Log:
```
The following FormatException was thrown building StoryblokRichText(dirty, dependencies: [InheritedCupertinoTheme, _InheritedTheme, _LocalizationsScope-[GlobalKey#75f2f]]):
Invalid radix-16 number (at character 1)

^

The relevant error-causing widget was: 
  StoryblokRichText StoryblokRichText:file:///C:/Users/PC/Projects/my_app/lib/ui/home/screens/home_screen.dart:166:22
When the exception was thrown, this was the stack: 
#0      int._handleFormatError (dart:core-patch/integers_patch.dart:127:5)
#1      int.parse (dart:core-patch/integers_patch.dart:51:14)
#2      HexColor._getColorFromHex (package:flutter_storyblok/src/widgets/hex_color.dart:10:16)
#3      new HexColor (package:flutter_storyblok/src/widgets/hex_color.dart:13:43)
#4      mapIfNotNull (package:flutter_storyblok/src/utils.dart:4:36)
#5      _RichTextLeafMarkableWidget.buildTextStyle (package:flutter_storyblok/src/widgets/rich_text.dart:153:9)
#6      _RichTextLeafMarkableWidget.buildTextSpan (package:flutter_storyblok/src/widgets/rich_text.dart:182:14)
#7      _RichTextLeafBuildWidget.buildRichText.<anonymous closure> (package:flutter_storyblok/src/widgets/rich_text.dart:131:51)
#8      MappedListIterable.elementAt (dart:_internal/iterable.dart:425:31)
#9      ListIterator.moveNext (dart:_internal/iterable.dart:354:26)
#10     new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
#11     new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
#12     new List.of (dart:core-patch/array_patch.dart:39:18)
#13     ListIterable.toList (dart:_internal/iterable.dart:224:7)
#14     _RichTextLeafBuildWidget.buildRichText (package:flutter_storyblok/src/widgets/rich_text.dart:142:16)
#15     _RichTextContainerWidget.buildContainerWidget (package:flutter_storyblok/src/widgets/rich_text.dart:60:71)
#16     StoryblokRichText.build.<anonymous closure> (package:flutter_storyblok/src/widgets/rich_text.dart:36:25)
#17     MappedListIterable.elementAt (dart:_internal/iterable.dart:425:31)
#18     ListIterator.moveNext (dart:_internal/iterable.dart:354:26)
#19     new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
#20     new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
#21     new List.of (dart:core-patch/array_patch.dart:39:18)
#22     ListIterable.toList (dart:_internal/iterable.dart:224:7)
#23     StoryblokRichText.build (package:flutter_storyblok/src/widgets/rich_text.dart:38:12)
#24     StatelessElement.build (package:flutter/src/widgets/framework.dart:5687:49)
#25     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5617:15)
#26     Element.rebuild (package:flutter/src/widgets/framework.dart:5333:7)
#27     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5599:5)
#28     ComponentElement.mount (package:flutter/src/widgets/framework.dart:5593:5)
...     Normal element mounting (4 frames)
#32     Element.inflateWidget (package:flutter/src/widgets/framework.dart:4468:16)
#33     MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:7035:36)
#34     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:7047:32)
...     Normal element mounting (144 frames)
#178    Element.inflateWidget (package:flutter/src/widgets/framework.dart:4468:16)
#179    Element.updateChild (package:flutter/src/widgets/framework.dart:3963:18)
#180    SliverMultiBoxAdaptorElement.updateChild (package:flutter/src/widgets/sliver.dart:967:37)
#181    SliverMultiBoxAdaptorElement.createChild.<anonymous closure> (package:flutter/src/widgets/sliver.dart:952:20)
#182    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:3038:19)
#183    SliverMultiBoxAdaptorElement.createChild (package:flutter/src/widgets/sliver.dart:944:12)
#184    RenderSliverMultiBoxAdaptor._createOrObtainChild.<anonymous closure> (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:361:23)
#185    RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:2719:59)
```

## Development Environment:
- Storyblok package: 1.0.4
- Flutter SDK: 3.24.0
- Both Visual Studio Code, IntelliJ IDEA was used

## Additional Context
I haven't tested this but running an iOS build.

**Comment by PhiFry** [comment]
@joshthedevelopa Interesting error. 

Could you export the story json data and share with me? 

How is this "switch functionality in mobile to enable switching and querying between two languages" implemented? 

Does it send a "de_DE" for the language parameter: `getStory(language: "de_DE")`?

**Comment by joshthedevelopa** [comment]
@PhiFry 
Unfornately, I can't reveal the contents of the story JSON data.
Yeah, I made a call using `getStory(language; "de")`  and I do get the German data as expected.

Alright, I see. Then I would have to inform the editors about it. I made a supplementary adjustment to cases as such in the pull request, a fallback color (Black) when the `int.parse` function falls.
Thanks for the insist as well.

**Comment by PhiFry** [comment]
I tried to replicate the issue and I think I succeeded. The issue isnt the color string being empty but rather it changes format from a `#RRGGBB` format to a `rgb(rrr, ggg, bbb)` format when I copied and pasted the rich text into the new language.

**Comment by PhiFry** [comment]
Fixed with #46

**Comment by PhiFry** [comment]
@joshthedevelopa Fix has been published as version 1.2.0 👍 Thank you for sharing this issue and contributing to making the package better!
